lua/conform/runner.lua

```diff
@@ -298,6 +298,10 @@ local last_run_errored = {}
 ---@param callback fun(err?: conform.Error, output?: string[])
 ---@return integer? job_id
 local function run_formatter(bufnr, formatter, config, ctx, input_lines, opts, callback)
+  vim.api.nvim_exec_autocmds("User", {
+    pattern = "ConformPreFormatter",
+    data = { formatter = formatter },
+  })
   log.info("Run %s on %s", formatter.name, vim.api.nvim_buf_get_name(bufnr))
   log.trace("Input lines: %s", input_lines)
   callback = util.wrap_callback(callback, function(err)
```

Comment:
We should try to provide the guarantee that if `ConformPreFormatter` is called, then `ConformPostFormatter` will also be called. Right now there are some error paths that won't hit the second autocmd. I believe if we move it into this `util.wrap_callback` function it will always get called.

Reply:
I've moved this into the callback and changed the event data to return the error (code and message)